tutorial.ipynb

```diff
@@ -490,20 +491,366 @@
+    "        terrorists[person]+=1\n",
+    "print(terrorists.most_common(10))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "#An alternative approach (inspired by http://spacy.io/docs/tutorials/twitter-filter) is shown\n",
+    "#here. However if you run this you will see all sentences are classified as positive\n",
+    "\n",
+    "from numpy import dot\n",
+    "from numpy.linalg import norm\n",
+    "\n",
+    "def classify_sentence(sentence):\n",
+    "    def get_vector(word):\n",
+    "        return nlp.vocab[word].repvec\n",
+    "    \n",
+    "    def cos(v1,v2):\n",
+    "        return dot(v1,v2) / (norm(v1)*norm(v2))\n",
+    "  \n",
+    "    sen = [w.repvec for w in sentence]\n",
+    "    accept = map(get_vector,'Terrorists'.split())\n",
+    "    reject = map(get_vector,'Civilians'.split())\n",
+    "    \n",
+    "    y = sum(max(cos(w1,w2),0) for w1 in sen for w2 in accept)\n",
+    "    n = sum(max(cos(w1,w2),0) for w1 in sen for w2 in reject)\n",
+    "    \n",
+    "    if (y/(y+n)) >= 0.5 or True:\n",
+    "        return True\n",
+    "    else:\n",
+    "        return False"
+   ]
+  },
+  {
    "cell_type": "code",
    "execution_count": null,
    "metadata": {
     "collapsed": false
    },
    "outputs": [],
-   "source": []
+   "source": [
+    "#Run sentence through classifier and print how many (of 30367) pass positive\n",
+    "\n",
+    "count = 0\n",
+    "for sen in sentences:\n",
+    "    if classify_sentence(sen) is True:\n",
+    "        count+=1\n",
+    "print(count)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "So we have shown two ways to distinguish terrorists from other persons in the text. Neither were successful. Maybe you can find a better way by analysing the syntactic dependencies?"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Plotting the number of terrorist incidents in a country over time"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "# Retrieve all the locations and store in a defaultdict object\n",
+    "locations = defaultdict(list)\n",
+    "for ent in terror_doc.ents:\n",
+    "    if ent.label_ == 'GPE':\n",
+    "        locations[ent.lemma_].append(ent.start)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "# Using the same procedure as for Pride and Prejudice we can extract the occurrences of a place over time.\n",
+    "# For simplicity, commonly mentioned places inside a country and have been added to the country count\n",
+    "\n",
+    "NUM_BINS = 40\n",
+    "\n",
+    "iraq_occurrences = normalize_occurences(locations['iraq']) \\\n",
```

Comment:
This function `normalize_occurences` was written very purely and I think it should not be used for here since normalization factor is static. Could you please rewrite `normalize_occurences ` function to accept normalization factor as well.